lib/html/pipeline/emoji_filter.rb

```diff
@@ -71,7 +72,36 @@ def asset_path(name)
 
       # Build an emoji image tag
       def emoji_image_tag(name)
-        "<img class='emoji' title=':#{name}:' alt=':#{name}:' src='#{emoji_url(name)}' height='20' width='20' align='absmiddle' />"
+        html_attrs =
+          default_img_attrs(name).
+            merge!(customized_img_attrs(name)).
+            except(*customized_img_attrs(name).select { |_, v| v.nil? }.keys).
+            map { |attr, value| %(#{attr}="#{value}") }.join(" ".freeze)
+
+        "<img #{html_attrs}>"
+      end
+
+      # Default attributes for img tag
+      def default_img_attrs(name)
+        {
+          "class" => "emoji".freeze,
+          "title" => ":#{name}:",
+          "alt" => ":#{name}:",
+          "src" => "#{emoji_url(name)}",
+          "height" => "20".freeze,
+          "width" => "20".freeze,
+          "align" => "absmiddle".freeze,
+        }
+      end
+
+      def customized_img_attrs(name)
+        @_customized_img_attrs ||= begin
+          return {} unless context[:img_attrs]
+
+          context[:img_attrs].each do |key, value|
+            context[:img_attrs][key] = value.call(name) if value.respond_to?(:call)
+          end
+        end
```

Comment:
This method can be inlined as well. Similar reasoning as before. Here, we're mutating the `context` hash by adding more elements to it. While that's doable, it makes it difficult to debug pipelines because it's unclear which filter modified a context. Instead, we can leave the context as passed in options and use it's values. I haven't tested this, but I imagine something like:

```ruby
default_attrs = default_img_attrs(name)

# This avoids creating an extra empty hash for each invocation and only merges if user passed something in
merged_attrs = context[:img_attrs] ? default_attrs.merge(context[:img_attrs]) : default_attrs

# This builds the attributes and handles procs. In your implementation, you remove nils, but I think that should be the responsibility of the caller. Nils would print as an empty value, which would still be valid.
html_attrs = merged_attrs.map do |k, v| 
  "#{k}='#{v.try(:call) || v}'"
end

"<img #{html_attrs.join(' '}>"
```